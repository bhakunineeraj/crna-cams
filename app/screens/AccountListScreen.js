import React from 'react';
import {
  Button,
  FlatList,
  Text,
  View,
  RefreshControl,
  ActivityIndicator
} from 'react-native';
import { ListItem, SearchBar } from 'react-native-elements';
import { connect } from 'react-redux';
import { getAccountList } from '../api/account';
import styles from '../styles';

class AccountListScreen extends React.Component {
  static navigationOptions = ({ navigation }) => {
    return {
      title: 'Accounts',
      headerRight: (
        <Button
          title="Map"
          onPress={() => navigation.navigate('accountListMap', null)}
        />
      )
    };
  };

  componentWillMount() {
    this._refreshAccountList();
  }

  _refreshAccountList = () => {
    this.setState({ loading: true });

    const { settings, auth } = this.props;
    const accessToken = settings.useMockData ? null : auth.userInfo.idToken;
    getAccountList(accessToken).then(value =>
      this.setState({ loading: false, accountList: value })
    );
  };

  _keyExtractor = (item, index) => item.id; // eslint-disable-line no-unused-vars

  _renderItem = ({ item }) =>
    <ListItem
      onPress={() =>
        this.props.navigation.navigate('accountDetail', { account: item })}
      title={item.name}
      subtitle={
        <View style={styles.subtitleView}>
          <Text style={styles.ratingText}>
            {item.notes}
          </Text>
        </View>
      }
    />;

  render() {
    return this.state.loading
      ? <ActivityIndicator size="large" />
      : <View style={styles.container}>
          <SearchBar noIcon lightTheme placeholder="" />
          <FlatList
            refreshControl={
              <RefreshControl
                refreshing={this.state.loading}
                onRefresh={this._refreshAccountList}
              />
            }
            data={this.state.accountList}
            keyExtractor={this._keyExtractor}
            renderItem={this._renderItem}
          />
        </View>;
  }
}

const mapStateToProps = ({ settings, auth }) => {
  return { settings, auth };
};

export default connect(mapStateToProps)(AccountListScreen);
