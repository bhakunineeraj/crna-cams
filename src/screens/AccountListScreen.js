import React from 'react';
import {
  Button,
  StyleSheet,
  FlatList,
  Text,
  View,
  RefreshControl
} from 'react-native';
import { ListItem, SearchBar } from 'react-native-elements';
import Loading from '../components/Loading';
import { getAccountList } from '../api/account';

export default class AccountListScreen extends React.Component {
  static navigationOptions = ({ navigation }) => {
    return {
      title: 'Accounts',
      headerRight: (
        <Button
          title="Map"
          onPress={() => navigation.navigate('accountListMap', null)}
        />
      )
    };
  };

  componentWillMount() {
    this._refreshAccountList();
  }

  _refreshAccountList = () => {
    this.setState({ loading: true });
    getAccountList().then(value =>
      this.setState({ loading: false, accountList: value })
    );
  };

  _keyExtractor = (item, index) => item.id;

  _renderItem = ({ item }) =>
    <ListItem
      onPress={() =>
        this.props.navigation.navigate('accountDetail', { account: item })}
      title={item.name}
      subtitle={
        <View style={styles.subtitleView}>
          <Text style={styles.ratingText}>
            {item.notes}
          </Text>
        </View>
      }
    />;

  render() {
    return this.state.loading
      ? <Text>loading...</Text>
      : <View style={styles.container}>
          <SearchBar noIcon lightTheme placeholder="" />
          <FlatList
            refreshControl={
              <RefreshControl
                refreshing={this.state.loading}
                onRefresh={this._refreshAccountList}
              />
            }
            data={this.state.accountList}
            keyExtractor={this._keyExtractor}
            renderItem={this._renderItem}
          />
        </View>;
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff'
  },
  getStartedText: {
    fontSize: 17,
    color: 'rgba(96,100,109, 1)',
    lineHeight: 24,
    textAlign: 'center'
  }
});
