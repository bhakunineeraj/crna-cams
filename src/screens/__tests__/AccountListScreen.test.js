import React from 'react';
import AccountListScreen from '../AccountListScreen';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const tree = renderer.create(<AccountListScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
