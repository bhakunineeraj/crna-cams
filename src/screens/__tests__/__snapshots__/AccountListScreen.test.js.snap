// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#393e42",
          "borderBottomColor": "#000",
          "borderBottomWidth": 1,
          "borderTopColor": "#000",
          "borderTopWidth": 1,
        },
        Object {
          "backgroundColor": "#e1e8ee",
          "borderBottomColor": "#e1e1e1",
          "borderTopColor": "#e1e1e1",
        },
        undefined,
      ]
    }
  >
    <TextInput
      placeholder=""
      placeholderTextColor="#86939e"
      style={
        Array [
          Object {
            "backgroundColor": "#303337",
            "borderRadius": 3,
            "color": "#86939e",
            "fontSize": 17.5,
            "height": 30,
            "margin": 8,
            "overflow": "hidden",
            "paddingLeft": 26,
            "paddingRight": 19,
          },
          Object {
            "backgroundColor": "#bdc6cf",
          },
          Object {
            "paddingLeft": 9,
          },
          false,
          undefined,
          undefined,
          false,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <RCTScrollView
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={undefined}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
